Bioinf_Utility/bioinf_utility.py

```diff
@@ -0,0 +1,173 @@
+
+    output_fastq (str, optional):
+        Name of the output file to which filtered sequences will be saved.
+        If not specified, the output file will be saved with the same base name
+        as the input file.
+        File will be saved to the 'filtered' subfolder in FASTQ format.
+
+    gc_bounds (tuple or int, optional):
+        A tuple specifying the minimum and maximum GC-content percentages
+        (default is (0, 100)). If an integer is provided, it is treated as the
+        maximum GC content with a minimum of 0.
+
+    length_bounds (tuple or int, optional):
+        A tuple specifying the minimum and maximum sequence lengths
+        (default is (0, 2**32)). If an integer is provided, it is treated as
+        the maximum sequence length with a minimum of 0.
+
+    quality_threshold (int, optional):
+        The minimum acceptable quality score for sequences (default is 0).
+        All sequences must have a quality score greater than or equal to this
+        threshold to pass the filter.
+
+    Returns:
+        None: Writes the filtered sequences to the specified output FASTQ file.
+        If any of the input parameters are invalid, returns the error message.
+
+    Example:
+        To filter sequences from a FASTQ file based on specific criteria:
+
+        filter_fastq(
+            "data/example_fastq.fastq",
+            "each_filter_output.fastq",
+            (40, 80),
+            (10, 20),
+            35
+        )
+
+        # This will write the filtered sequences to
+        # 'filtered/each_filter_output.fastq'.
+    """
+    if type(gc_bounds) is int and gc_bounds > 0:
+        gc_bounds = (0, gc_bounds)
+    elif type(gc_bounds) is not tuple and len(gc_bounds) != 2:
+        return "Wrong value for the gc_bounds argument"
+
+    if type(length_bounds) is int and length_bounds > 0:
+        length_bounds = (0, length_bounds)
+    elif type(length_bounds) is not tuple and len(length_bounds) != 2:
+        return "Wrong value for the length_bounds argument"
+
+    if type(quality_threshold) is not int or quality_threshold < 0:
+        return "Wrong value for the quality_threshold argument"
+
+    if output_fastq is None:
+        output_filtered_path = os.path.join(
+            "filtered", os.path.basename(input_fastq)
+        )
+    else:
+        output_filtered_path = os.path.join("filtered", output_fastq)
+
+    if not os.path.exists("filtered"):
+        os.mkdir("filtered")
+
+    with (
+        open(input_fastq, "r") as input_file,
+        open(output_filtered_path, "w") as output_file,
+    ):
+        seq_name, sequence, quality_seq = read_fastq_seq_from_file(input_file)
+        while seq_name:
+            if (
+                is_seq_gc_in_bounds(sequence, gc_bounds)
+                and is_seq_len_in_bounds(sequence, length_bounds)
+                and is_seq_quality_higher_than_threshold(
+                    quality_seq, quality_threshold
+                )
```

Comment:
Сложное условие, тяжело читается стороннему человечку. Мб лучше каждую часть условия вынести в отдельную переменную, а потом уже запихать в условие:

```suggestion
    is_gc_valid = is_seq_gc_in_bounds(sequence, gc_bounds)
    is_len_valid = is_seq_len_in_bounds(sequence, length_bounds)
    is_quality_valid = is_seq_quality_higher_than_threshold(quality_seq, quality_threshold)
    if is_gc_valid and is_len_valid and is_quality_valid:
```